modules/core/src/main/java/org/locationtech/jts/geom/LineSegment.java

```diff
@@ -505,16 +505,49 @@ public Coordinate reflect(Coordinate p) {
    */
   public Coordinate closestPoint(Coordinate p)
   {
-    double factor = projectionFactor(p);
-    if (factor > 0 && factor < 1) {
-      return project(p, factor);
+    return closestPoint(p, false);
+  }
+  
+  /**
+   * Computes the closest point on this line segment to another point.
+   * @param p the point to find the closest point to
+   * @param skip0 If true, do not check p0. Client will have determined this is redundant.
+   * @return a Coordinate which is the closest point on the line segment to the point p
+   */
+  public Coordinate closestPoint(Coordinate p, boolean skip0)
```

Comment:
Is there a use case for this method?

Reply:
See the caller in the pull request. For the DistanceOp point-linesegment calculation, each linesegment shares an endpoint. The existing code checks each linesegment in sequence and possibly both endpoints of the linesegment. So the shared endpoints could be checked twice, unnecessarily.

Reply:
How much of a performance improvement does this create?

Reply:
It's probably better to localize this method to the `DistanceOp` class